Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverFee.hs

```diff
@@ -554,14 +618,28 @@ updateRegisterationFeeStatusByDriverId :: (MonadFlow m, EsqDBFlow m r, CacheFlow
 updateRegisterationFeeStatusByDriverId status (Id driverId) = do
   now <- getCurrentTime
   updateOneWithKV
-    [Se.Set BeamDF.status status, Se.Set BeamDF.updatedAt now]
+    ( [Se.Set BeamDF.status status, Se.Set BeamDF.updatedAt now]
+        <> [Se.Set BeamDF.badDebtDeclarationDate $ Just now | status `elem` [EXEMPTED, INACTIVE]]
+    )
     [Se.And [Se.Is BeamDF.driverId (Se.Eq driverId), Se.Is BeamDF.feeType (Se.Eq MANDATE_REGISTRATION), Se.Is BeamDF.status (Se.Eq PAYMENT_PENDING)]]
 
 updateCollectedPaymentStatus :: (MonadFlow m, EsqDBFlow m r, CacheFlow m r) => DriverFeeStatus -> Maybe Text -> UTCTime -> Id DriverFee -> m ()
 updateCollectedPaymentStatus status volunteerId now (Id driverFeeId) = do
+  utcNow <- getCurrentTime
   updateOneWithKV
-    [Se.Set BeamDF.status status, Se.Set BeamDF.updatedAt now, Se.Set BeamDF.collectedBy volunteerId, Se.Set BeamDF.collectedAt (Just now)]
+    ( [Se.Set BeamDF.status status, Se.Set BeamDF.updatedAt now, Se.Set BeamDF.collectedBy volunteerId, Se.Set BeamDF.collectedAt (Just now)]
+        <> [Se.Set BeamDF.badDebtDeclarationDate $ Just now | status `elem` [EXEMPTED, INACTIVE]]
+    )
     [Se.Is BeamDF.id (Se.Eq driverFeeId)]
+  updateOneWithKV
```

Comment:
if both updates are separately required we can make it a separate function and do.

```
[Se.Set BeamDF.badDebtRecoveryDate $ Just utcNow]
    [ Se.And
        [ Se.Is BeamDF.id (Se.Eq driverFeeId),
          Se.Is BeamDF.status $ Se.In [COLLECTED_CASH, CLEARED],
          Se.Is BeamDF.badDebtDeclarationDate $ Se.Not (Se.Eq Nothing),
          Se.Is BeamDF.badDebtRecoveryDate $ Se.Eq Nothing
        ]
    ]
```

This is redundant in functions above